imgui-SFML.cpp

```diff
@@ -1403,4 +1676,183 @@ void updateMouseCursor(sf::Window& window) {
     }
 }
 
+
+#ifdef VIEWPORTS_ENABLE
+
+void SFML_CreateWindow(ImGuiViewport* viewport) {
+#if SFML_VERSION_MAJOR >= 3
+    sf::RenderWindow* window = new sf::RenderWindow(sf::VideoMode({(unsigned)viewport->Size.x,
+                                                                   (unsigned)viewport->Size.y}),
+                                                    "", sf::Style::None);
+#else
+    sf::RenderWindow* window =
+        new sf::RenderWindow(sf::VideoMode(viewport->Size.x, viewport->Size.y), "",
+                             sf::Style::None);
+#endif
+    window->setVisible(false);
+    WindowContext* data = IM_NEW(WindowContext)(window, ImGui::GetCurrentContext(), true);
+    data->windowHasFocus = true;
+    viewport->PlatformUserData = data;
+    viewport->PlatformHandle = window;
+}
+
+void SFML_DestroyWindow(ImGuiViewport* viewport) {
+    if (WindowContext* wc = (WindowContext*)viewport->PlatformUserData) {
+        if (!wc->imContextOwner) {
+            wc->window->close();
+            IM_DELETE(wc);
+        }
+    }
+    viewport->PlatformUserData = nullptr;
+    viewport->PlatformHandle = nullptr;
+}
+
+void SFML_ShowWindow(ImGuiViewport* viewport) {
+    WindowContext* wc = (WindowContext*)viewport->PlatformUserData;
+    wc->window->setVisible(true);
+}
+
+void SFML_SetWindowPos(ImGuiViewport* viewport, ImVec2 pos) {
+    WindowContext* wc = (WindowContext*)viewport->PlatformUserData;
+    wc->window->setPosition(pos);
+}
+
+ImVec2 SFML_GetWindowPos(ImGuiViewport* viewport) {
+    WindowContext* wc = (WindowContext*)viewport->PlatformUserData;
+
+#ifdef _WIN32    
+    RECT clientAreaRect;
+    HWND hwnd = wc->window->getSystemHandle();
+    GetClientRect(hwnd, &clientAreaRect);
+    MapWindowPoints(hwnd, NULL, (LPPOINT)&clientAreaRect, 2);
+    return { (float)clientAreaRect.left, (float)clientAreaRect.top };
+#else
+    if (wc->imContextOwner) return wc->window->getPosition() + sf::Vector2i(0, 24);
```

Comment:
What is this (0, 24) offset?

Reply:
This was an early attempt to cancel out that title bar offset in the main viewport. I will delete this.